@ecomplus/storefront-components/src/js/DiscountApplier.js

```diff
@@ -113,6 +89,33 @@ export default {
       this.localAmountTotal = (amount.subtotal || 0) + (amount.freight || 0)
     },
 
+    addFreebieItems (ecomCart, productIds) {
+      if (Array.isArray(productIds)) {
+        productIds.forEach(productId => {
+          const itemFreebie = ecomCart.data.items.find(item => item.product_id === productId)
+          if (!itemFreebie || !this.freebieCount) {
```

Comment:
porque o check por esse `freebieCount`?

Reply:
ele pode ter mais de um brinde na real

Reply:
e sem o `this.freebieCount` o `addFreebieItems` também pode ser matido fora da instância do componente, como uma função externa em vez de método

Reply:
Porque ele gera uma inserção infinita. O addFreebieItems é chamado toda vez que tenho update no carrinho, que é o certo, ou seja, toda vez que insiro o brinde, o carrinho atualiza e insere de novo e vai assim infinitamente. Como o produto já existe, acredito que ele só aumenta a quantidade e não entra flag de freebie pra travar a próxima inserção.

Reply:
Por isso tinha pensado em jogar com um _id diferente, relacionar via product_id prop e nele conseguir colocar a flag